runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/PTransformMatchers.java

```diff
@@ -158,4 +160,39 @@ public boolean matches(AppliedPTransform<?, ?, ?> application) {
       }
     };
   }
+
+  /**
+   * A {@link PTransformMatcher} which matches a {@link ParDo.Bound} or {@link ParDo.BoundMulti}
+   * where the {@link DoFn} is of the provided type.
+   */
+  public static PTransformMatcher parDoWithFnType(final Class<? extends DoFn> fnType) {
+    return new PTransformMatcher() {
+      @Override
+      public boolean matches(AppliedPTransform<?, ?, ?> application) {
+        DoFn<?, ?> fn;
+        if (application.getTransform() instanceof ParDo.Bound) {
+          fn = ((ParDo.Bound) application.getTransform()).getFn();
```

Comment:
That's accurate. Theoretically for well-known DoFns with a useful URN a similar matcher could be constructed